### DISCUSSION TITLE
Pydantic Models for multi var read

### DISCUSSION THREAD

**Discussion by lautarodapin** [gijzelaerr/python-snap7#256] [Ideas]
What about using Pydantic models or dataclasses to integrate with other APIs?
I have in produccion a script with `python-snap7` that would benefice by using this new library.
Also it will make easier to use the multivar reading.

**Comment by gijzelaerr** [comment]
How do you envision such integration? For simplicity and backwards compatibility, I think it is good to keep the library simple, but we can always add functions layers.

**Reply by lautarodapin** [reply]
how would you add those functions? in the same library? or in another lib?

**Reply by gijzelaerr** [reply]
in a submodule, and have the potential dependencies in an optional setup target. We can always move the submodule to a new project if it becomes too large or out of scope.

**Reply by lautarodapin** [reply]
Ok, I would like to extend the `DB` class to have a better potencial, and use it for multi read different DBs marks or inputs.
I would work on that, thannks
